scripts/sweep_common.py

```
@@ -333,6 +336,42 @@ def predict_bo_budget(smiles: str) -> Optional[int]:
+    skip and just run it).
+
+    Uses bouquet's own ``detect_dihedrals`` so the count matches what the CLI
+    sees under ``--auto``. Imported lazily so the analyze/traj paths never pay the
+    rdkit import.
+    """
+    try:
+        from bouquet.setup import detect_dihedrals, get_initial_structure
+
+        _, mol = get_initial_structure(smiles)
+        return len(detect_dihedrals(mol))
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)